### ISSUE TITLE
Request for CMake for fizmo to multi-head Glk libraries

### ISSUE THREAD

**Issue by WakeRealityDev** [chrender/fizmo#23]
Hello. It's great to see fizmo have so many front-end options and to continue to be updated. 

I'd really like to see exactly what source files are needed and a recommended CMakeList.txt to allow fizmo to be used in Gargoyle and be compiled as part of a set of multiple interpreters.  A new project has been posted to split all the 'terps backends and a common set of CMake files to link different Glk systems such as RemGlk, CheapGlk, etc. https://github.com/cspiegel/terps

Any interest is appreciated. Thank you.

**Comment by chrender** [comment]
Hi, and many thanks for your interest in adding fizmo to this project.

I've got to admit that I haven't got a clue about cmake, but I'm willing to give it a try if it helps to include fizmo. Howver from what I understood so far it appears that the main challenge in adding fizmo could be the fact that fizmo's functionality is separated into multiple static libraries. Would it be possible to reference external libraries during the build? In that case we'd only need a single file with just a few lines of code – example: https://github.com/chrender/fizmo-glktermw/blob/6bcf0b6/src/fizmo-glktermw/fizmo-glktermw.c – to connect glk-fizmo to an arbitrary GLK fronend.

**Comment by WakeRealityDev** [comment]
Hello. One of the main advantages of CMake is that it is the basis of cross-compiler chains for Android mobile phones (Google's Android Studio), Qt Creator targets, etc. This includes ARM, MIPS and other CPU besides just Intel - so typically everything has to be compiled several times in a row.

I think what we typically need to make a fresh CMake file is a flat list of all the .c and .h files in the proper order.  In other words, all the interpreter source files up to and including the fizmo-glktermw.c   - or we could split each static library into a set of source file lists.

By a a "flat list" - I mean we basically remove all the if conditions. We don't need your SDL2 files, etc. But basically source source file so that RemGlk or any other GLK front-end could be linked.

**Comment by chrender** [comment]
Aiming for a pure CMake solution is certainly possible. This will however take some time. As I see it, in this case the only viable solution would be to fully support CMake in parallel with Autotools for the two libraries which we'll need to create a glk-fizmo, which means that I'll have to learn about CMake in such a way that I can support it and that in the long term that I would have to entirely switch from Autotools to CMake, since supporting two build systems side by side doesn't make too much sense – even supporting a single one for now twelve modules isn't much fun.

As for now, I'd put this issue on hold until fizmo 0.8.5 has been finished – which I hope will happen in the current or the next week – as well as packaged and distributed. After that, and depending on the bugs and problems from the new 0.8.5 release, I'l try to get a grasp on CMake.
